### PR TITLE
Use Lightning's throughput

### PR DESCRIPTION
**Pull request by carmocca**
See https://github.com/Lightning-AI/lightning/pull/18848

### CODE REVIEW

File: finetune/adapter.py [hunk 1]
```diff
@@ -150,10 +147,12 @@ def train(
         # this assumes that all samples have a fixed length equal to the longest sequence length
         # which is most likely false during finetuning
         x = torch.randint(0, 1, (micro_batch_size, longest_seq_length))
-        measured_flops = measure_flops(meta_model, x)
+        forward_fn = lambda: meta_model(x)
+        measured_flops = measure_flops(meta_model, forward_fn, torch.Tensor.sum)
```
Comment (awaelchli) [inline finetune/adapter.py:151]: Shouldn't you include the loss as well? For example, I did it like this:
https://github.com/Lightning-AI/lit-gpt/pull/677/files#diff-00fedfb359238295e8b95bc7c9f90975463bf4c40f03b2a010a118ec21f2f8d3R163

File: lit_gpt/utils.py [hunk 2]
```diff
@@ -309,3 +312,36 @@ def load_checkpoint(fabric: L.Fabric, model: nn.Module, checkpoint_path: Path, s
...
+def flops_per_param(max_seq_length: int, n_layer: int, n_embd: int, n_params: int) -> int:
+    flops_per_token = 2 * n_params  # each parameter is used for a MAC (2 FLOPS) per network operation
+    # this assumes that all samples have a fixed length equal to the block size
+    # which is most likely false during finetuning
+    flops_per_seq = flops_per_token * max_seq_length
+    attn_flops_per_seq = n_layer * 2 * 2 * (n_embd * (max_seq_length**2))
+    return flops_per_seq + attn_flops_per_seq
+
+
+def estimate_flops(model: "GPT") -> int:
```
Comment (awaelchli) [inline lit_gpt/utils.py:326]:
```suggestion
def estimate_flops(model: "GPT", training: bool) -> int:
```
IMO we should ask for training specifically rather than using model.training

**Review by awaelchli — APPROVED** [review]
